src/libphonenumber/PhoneNumber.php

```diff
@@ -546,13 +546,17 @@ public function unserialize($serialized)
     {
         $data = unserialize($serialized);
 
-        $this->setCountryCode($data[0]);
-        $this->setNationalNumber($data[1]);
-        $this->setExtension($data[2]);
-        $this->setItalianLeadingZero($data[3]);
-        $this->setNumberOfLeadingZeros($data[4]);
-        $this->setRawInput($data[5]);
-        $this->setCountryCodeSource($data[6]);
-        $this->setPreferredDomesticCarrierCode($data[7]);
+        list (
+            $this->countryCode,
+            $this->nationalNumber,
+            $this->extension,
+            $this->italianLeadingZero,
+            $this->numberOfLeadingZeros,
+            $this->rawInput,
+            $this->countryCodeSource,
+            $this->preferredDomesticCarrierCode
+        ) = $data;
+
+        $this->hasNumberOfLeadingZeros = true;
```

Comment:
Lets set `$hasNumberOfLeadingZeros = true` when `$numberOfLeadingZeros != 0`, (shouldn't this be 0, rather than 1?) and leave it out of the serialised array (saves a bit of space there).